pkg/azureutils/azure_disk_utils.go

```diff
@@ -209,28 +209,33 @@ func CreateValidDiskName(volumeName string, usedForLabel bool) string {
 func GetCloudProviderFromClient(kubeClient *clientset.Clientset, secretName string, secretNamespace string, userAgent string) (*azure.Cloud, error) {
```

Comment:
Since the V1 and V2 now share this function, we might, unfortunately, need to create a new `GetCloudProviderFromClient` for V2 to aid with merging.

Reply:
Or just deal with the merge issues. Long term, I plan to offer my `cloud-provider-azure` changes to the upstream repo. If they take them, we can update the v1 initialization to match. I really want to get rid of the `cloud-provider-azure` dependency for GA, so it will likely not be an issue either way.

Reply:
That sounds good too, as long as we don't create the unintended duplicate during the merge 😄